src/network_kernels.cu

```diff
@@ -747,3 +747,90 @@ float *network_predict_gpu(network net, float *input)
     //cuda_free(state.input);   // will be freed in the free_network()
     return out;
 }
+
+float *network_predict_gpu_gl_texture(network net, uint32_t texture_id)
+{
+    if (net.gpu_index != cuda_get_device())
+        cuda_set_device(net.gpu_index);
+    int size = get_network_input_size(net) * net.batch;
+
+    // Map the OpenGL texture resource so CUDA can access it.
+    cudaGraphicsResource_t graphics_resource = NULL;
+    unsigned int flags = cudaGraphicsRegisterFlagsReadOnly;
+    CHECK_CUDA(cudaGraphicsGLRegisterImage(&graphics_resource, texture_id, GL_TEXTURE_2D, flags));
+    CHECK_CUDA(cudaGraphicsMapResources(1, &graphics_resource, 0));
```

Comment:
While maybe not _ideal_ to register and unregister textures on every frame, the alternative was keeping some sort of map of texture IDs to resources, which felt more intrusive and increases complexity in this PR.